OpenRA.Mods.Common/Traits/World/Locomotor.cs

```diff
@@ -431,7 +436,11 @@ void UpdateCellCost(CPos cell)
 
 			var cache = cellsCost[cell.Layer];
 
-			cache[cell] = cost;
+			var uv = cell.ToMPos(world.Map);
+			var oldCost = cache[uv];
+			cache[uv] = cost;
+
+			CellCostChanged?.Invoke(cell, oldCost, cost);
```

Comment:
Would there be any performance benefits to check if CellCostChanged is null and then dont access the oldCost?

Reply:
A minor benefit. Have updated the code to account for that.